### PR TITLE
Fix @warn deprecation and runtests for v1.0.0

### LINKED ISSUE #372: Julia 1.0.0 -> Error "UndefVarError: warn not defined"
The following line should use `@warn` instead of `warn`.

https://github.com/JuliaPy/PyPlot.jl/blob/c16484b03a666ead097c81ceda0966d5b332780b/src/init.jl#L160

Installation of `PyPlot` fails from command-line as `warn` is not supported anymore under 1.0.

```julia
Precompiling package PyPlot
WARNING: could not import Base.mimewritable into PyPlot
ERROR: LoadError: InitError: UndefVarError: warn not defined
Stacktrace:
 [1] find_backend(::PyCall.PyObject) at /opt/juliadepot/packages/PyPlot/jXCXB/src/init.jl:160
 [2] __init__() at /opt/juliadepot/packages/PyPlot/jXCXB/src/init.jl:182
 [3] _include_from_serialized(::String, ::Array{Any,1}) at ./loading.jl:627
 [4] _require_from_serialized(::String) at ./loading.jl:678
 [5] macro expansion at ./logging.jl:317 [inlined]
 [6] _require(::Base.PkgId) at ./loading.jl:941
 [7] require(::Base.PkgId) at ./loading.jl:852
 [8] macro expansion at ./logging.jl:311 [inlined]
 [9] require(::Module, ::Symbol) at ./loading.jl:834
during initialization of module PyPlot
in expression starting at /tmp/install_julia_10_packages.jl:44
```

### PR DESCRIPTION
**Pull request by asbisen**
some fixes to make PyPlot.jl install from command line and runtests.jl work on 1.0

Fixes #372

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+VERSION >= v"0.7.0" && using Base64
 
 if isdefined(Base, :Test) && !Base.isdeprecated(Base, :Test)
     using Base.Test
 else
     using Test
 end
 
-plot(1:5, 2:6, "ro-")
+plot(collect(1:5), collect(2:6), "ro-")
```
Comment (stevengj) [inline test/runtests.jl]: This change shouldn't be needed — we are hitting a bug (JuliaPy/PyCall.jl#533) in PyCall that should hopefully be fixed soon.

File: test/runtests.jl [hunk 2]
```diff
@@ -22,16 +25,33 @@ s = stringmime("application/postscript", fig);
...
-                     norm=PyPlot.Normalize01)["to_rgba"], PyArray, a)
-@test rgba ≈ [ 0.267004  0.004874  0.329415  1.0
-               0.229739  0.322361  0.545706  1.0
-               0.127568  0.566949  0.550556  1.0
-               0.369214  0.788888  0.382914  1.0
-               0.993248  0.906157  0.143936  1.0 ]
+if VERSION < v"0.7.0"
+	a = linspace(0,1,5)
+else
+	a = range(0,step=1,length=5)
```
Comment (stevengj) [inline test/runtests.jl]: can't we just call `Compat.range` in all versions?

Comment (stevengj) [reply test/runtests.jl]: The `range` call is not correct.  The analogue of the linspace call is `range(0, stop=1, length=5)`

Comment (stevengj) [reply test/runtests.jl]: Actually we can just do `0.0:0.25:1.0` in all versions.

File: test/runtests.jl [hunk 3]
```diff
@@ -22,16 +25,33 @@ s = stringmime("application/postscript", fig);
...
+end
+
+
+#
+# Not sure what this test is doing but
+# the results are different between 0.6 and 1.0
```
Comment (stevengj) [inline test/runtests.jl]: That's because `a` was wrong in 1.0 — once you fix the `range` call as noted above it should work again, I hope.

Comment (asbisen) [reply test/runtests.jl]: aah, I will make the change and resubmit. Thanks

File: test/runtests.jl (hunk 2 above)
Comment (asbisen) [reply test/runtests.jl]: agreed will be much cleaner

File: test/runtests.jl (hunk 1 above)
Comment (asbisen) [reply test/runtests.jl]: so should i revert this line back, or keep it this way..

Comment (stevengj) [reply test/runtests.jl]: revert it

Comment (stevengj) [reply test/runtests.jl]: (Fix merged, will be tagged in https://github.com/JuliaLang/METADATA.jl/pull/17187)

File: src/init.jl [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # PyPlot initialization — the hardest part is finding a working backend.
 using VersionParsing
+using Compat
```
Comment (stevengj) [inline src/init.jl]: This line is not needed — Compat is already imported in PyPlot.jl, from which `init.jl` is included.

**Pull request merged**